src/commands/registry.rs

```diff
@@ -61,11 +61,31 @@ impl Push {
         let app = spin_loader::local::from_file(&app_file, Some(dir.path())).await?;
 
         let mut client = spin_oci::Client::new(self.insecure, None).await?;
-        let digest = client.push(&app, &self.reference).await?;
 
+        let pb = ProgressBar::new_spinner();
+        pb.enable_steady_tick(Duration::from_millis(200));
+        pb.set_style(
+            ProgressStyle::with_template("{msg}{spinner}")
+                .unwrap()
+                .tick_strings(&[".", "..", "...", "....", "....."]),
+        );
+        pb.set_message("Pushing app to registry");
+
+        let digest = client.push(&app, &self.reference).await?;
```

Comment:
Do you need to do anything to the progress bar if this returns Err, or will it tidy itself up correctly as it goes out of scope?  (Asking because you explicitly handle the Err case in pull.)

Reply:
Good point. It tidies itself up when it goes out of scope. I think we can remove handling the Err case in pull.